docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,479 @@
+
+Outside of `Main()`, create a method called `EvaluateModel()`:
+```csharp
+public static void EvaluateModel(MLContext mlcontext, IDataView testDataView, ITransformer model)
+{
+
+
+}
+```
+
+Define the method by adding the following code to `EvaluateModel()`:
+
+[!code-csharp[EvaluateModel](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModel "Evaluate the model and print metrics")]
+
+The [Transform()](xref:Microsoft.ML.ITransformer.Transform%2A) method makes predictions for multiple provided input rows of a test dataset. Once you have the prediction set, you call the [Evaluate()](xref:Microsoft.ML.RecommendationCatalog.Evaluate%2A) method to assess the model, which compares the predicted values with the actual `Labels` in the test dataset and returns metrics on how the model is performing.
+
+The print statements display your evaluation metrics in the console, which should look similar to the following:
+
+```console
+=============== Training the model ===============
+iter      tr_rmse          obj
+   0       1.5390   3.1236e+05
+   1       0.9225   1.6025e+05
+   2       0.8690   1.5031e+05
+   3       0.8410   1.4575e+05
+   4       0.8141   1.4236e+05
+   5       0.7852   1.3909e+05
+   6       0.7546   1.3607e+05
+   7       0.7260   1.3350e+05
+   8       0.6988   1.3102e+05
+   9       0.6744   1.2936e+05
+  10       0.6530   1.2766e+05
+  11       0.6353   1.2638e+05
+  12       0.6201   1.2542e+05
+  13       0.6060   1.2464e+05
+  14       0.5955   1.2390e+05
+  15       0.5864   1.2329e+05
+======= Evaluating the model =======
+Rms: 0.981309770322365
+RSquared: 0.427519426849609
+```
+
+In this output, there are 20 iterations. In each iteration, the measure of error decreases and converges closer and closer to 0.
+
+The `root of mean squared error` (RMS or RMSE) is frequently used to measure the differences between values predicted by a model and the values actually observed in a test dataset. Technically it is the square root of the average of the squares of the errors. You want your RMSE score to be as close to 1 as possible.
+
+`R Squared` is the variation percentage in the predicted values explained by your model. It's a value between 0 and 1, and the closer the value is to 0, the better the model is.
+
+Add the following as the next line of code in the `Main()` method to call your `EvaluateModel()` method:
+
+[!code-csharp[EvaluateModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModelMain "Add EvaluateModel method in Main")]
+
+
+## Use your model
+
+Now you can use your trained model to make predictions on new data.
+
+Outside of `Main()`, create a method called `UseModelForSinglePrediction()`:
+```csharp
+public static void UseModelForSinglePrediction(MLContext mlcontext, ITransformer model)
+{
+
+
+}
+```
+
+Define the method by adding the following code to `UseModelForSinglePrediction()`:
```

Comment:
Change to "Use  the `PredictionEngine` to predict the rating by adding the following code to `UseModelForSinglePrediction()`: ".